### ISSUE TITLE
Sampling on three channel images

### ISSUE THREAD

**Issue by LV901MK2** [mazurowski-lab/segmentation-guided-diffusion#27]
First of all, thank you very much for your work.

I experience some issues on sampling images from a segmentation guided model. I have trained a model from scratch with retinal fundus images and one channel masks with four discrete classes labeled zero through three. 

But when sampling the images, they come out very colorful. Though details are captured nicely, the colors are off. 

I have trained the model like this:
```
CUDA_VISIBLE_DEVICES=0 python3 main.py \
    --mode train \
    --img_size 256 \
    --num_img_channels 3 \
    --dataset eyepacs \
    --img_dir /path/seg2img_dataset/data \
    --train_batch_size 16 \
    --eval_batch_size 8 \
    --num_epochs 100 \
    --seg_dir /path/seg2img_dataset/masks \
    --segmentation_guided \
    --num_segmentation_classes 4
```

These ares the samples I get for some test masks after 400 epochs:

<img width="1024" height="512" alt="Image" src="https://github.com/user-attachments/assets/c7cd2bc8-78f6-4fc6-bd1a-ff8b26eeff8a" />

corresponding masks:

<img width="1034" height="518" alt="Image" src="https://github.com/user-attachments/assets/06618d58-ab56-4c3d-8cc4-a79a7f334c9a" />

Expected result/original images:

<img width="1034" height="518" alt="Image" src="https://github.com/user-attachments/assets/f3866e20-177a-4a5a-bdf0-429761ca4a65" />

I have already tried changing up the `transforms.Normalize()` to use the real mean and std of my images

```
    if not load_images_as_np_arrays:
        preprocess = transforms.Compose(
            [
                transforms.Resize((config.image_size, config.image_size)),
                # transforms.RandomHorizontalFlip(), # flipping wouldn't result in realistic images
                transforms.ToTensor(),
                transforms.Normalize(
                    mean=[0.6500720977783203, 0.42943304777145386, 0.2655245363712311],
                    std=[0.2279864251613617, 0.1947232037782669, 0.17444148659706116])
            ]
        )
    else:
        # resizing will be done in the transform function
        preprocess = transforms.Compose(
            [
                transforms.Normalize(
                    mean=[0.6500720977783203, 0.42943304777145386, 0.2655245363712311],
                    std=[0.2279864251613617, 0.1947232037782669, 0.17444148659706116])
            ]
        )
```
But that, unfortunately, did not fix the problem either.

Do you have any idea, what is happening?
